tests/ops/op_math/test_parametrized_evolution.py

```
@@ -0,0 +1,242 @@
+
+        assert ev.H is H
+        assert ev.dt is None
+        assert qml.math.allequal(ev.t, [0, 2])
+        assert qml.math.allequal(ev.h_params, [1, 2])
+
+        assert ev.wires == H.wires
+        assert ev.num_wires == AnyWires
+        assert ev.name == "ParametrizedEvolution"
+        assert ev.id is None
+        assert ev.queue_idx is None
+
+        assert ev.data == []
+        assert ev.parameters == []
+        assert ev.num_params == 0
+
+    def test_has_matrix_true_via_op_have_matrix(self):
+        """Test that a parametrized evolution of operators that have `has_matrix=True`
+        has `has_matrix=True` as well."""
+
+        ops = [qml.PauliX(wires=0), qml.RZ(0.23, wires="a")]
+        coeffs = [1, 1]
+        H = ParametrizedHamiltonian(coeffs, ops)
+        ev = ParametrizedEvolution(H=H, params=[], t=0)
+
+        assert ev.has_matrix is True
+
+    def test_has_matrix_true_via_factor_has_no_matrix_but_is_hamiltonian(self):
+        """Test that a product of operators of which one does not have `has_matrix=True`
+        but is a Hamiltonian has `has_matrix=True`."""
+
+        ops = [qml.Hamiltonian([0.5], [qml.PauliX(wires=1)]), qml.RZ(0.23, wires=5)]
+        coeffs = [1, 1]
+
+        H = ParametrizedHamiltonian(coeffs, ops)
+        ev = ParametrizedEvolution(H=H, params=[], t=0)
+
+        assert ev.has_matrix is True
+
+    @pytest.mark.parametrize(
+        "first_factor", [qml.PauliX(wires=0), qml.Hamiltonian([0.5], [qml.PauliX(wires=1)])]
+    )
+    def test_has_matrix_false_via_factor_has_no_matrix(self, first_factor):
+        """Test that a product of operators of which one does not have `has_matrix=True`
+        has `has_matrix=False`."""
+
+        ops = [first_factor, MyOp(0.23, wires="a")]
+        coeffs = [1, 1]
+
+        H = ParametrizedHamiltonian(coeffs, ops)
+        ev = ParametrizedEvolution(H=H, params=[], t=0)
+
+        assert ev.has_matrix is False
```

Shouldnt we raise an error when an operator is passed without a matrix?

---

Done! But wouldn't it make more sense to raise the error in the `ParametrizedHamiltonian` class?